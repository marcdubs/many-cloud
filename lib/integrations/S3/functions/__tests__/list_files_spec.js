describe("list_files", () => {
  const expected_resolution = "watup fam";
  let data, func;

  beforeEach(() => {
    data = {
      s3: {
        listObjects: jest.fn().mockImplementation((params, callback) => {
          callback(null, expected_resolution);
        })
      }
    };
    func = require("../list_files")(data);
  });

  describe("with no folderID", () => {
    let actual;
    beforeEach(done => {
      func().then(res_ => {
        actual = res_;
        done();
      });
    });

    it("calls data.s3.listObjects", () => {
      expect(data.s3.listObjects).toHaveBeenCalled();
    });

    it("passes in correct default parameters", () => {
      expect(data.s3.listObjects.mock.calls[0][0]).toMatchSnapshot();
    });

    it("returns expected resolution", () => {
      expect(actual).toEqual(expected_resolution);
    });
  });

  describe("with folderID", () => {
    let actual;
    beforeEach(done => {
      func("id").then(res_ => {
        actual = res_;
        done();
      });
    });

    it("calls data.client.folders.getItems", () => {
      expect(data.s3.listObjects).toHaveBeenCalled();
    });

    it("passes in correct parameters", () => {
      expect(data.s3.listObjects.mock.calls[0][0]).toMatchSnapshot();
    });

    it("returns expected resolution", () => {
      expect(actual).toEqual(expected_resolution);
    });
  });

  describe("when listObjects throws an error", () => {
    const expected_error = "NOOOOOO";
    beforeEach(() => {
      data.s3.listObjects.mockImplementation((params, callback) => {
        callback(new Error(expected_error), null);
      });
    });

    it("rejects with the error", async () => {
      await expect(func("id")).rejects.toThrowErrorMatchingSnapshot();
    });
  });
});
