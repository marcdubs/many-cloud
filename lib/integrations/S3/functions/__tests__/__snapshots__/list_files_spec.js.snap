// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list_files when listObjects throws an error rejects with the error 1`] = `"NOOOOOO"`;

exports[`list_files with folderID passes in correct parameters 1`] = `
Object {
  "Bucket": undefined,
  "Delimiter": "/",
  "MaxKeys": 1000000,
  "Prefix": "id/",
}
`;

exports[`list_files with no folderID passes in correct default parameters 1`] = `
Object {
  "Bucket": undefined,
  "Delimiter": "/",
  "MaxKeys": 1000000,
}
`;
